docs/k0sctl-install.md

```diff
@@ -0,0 +1,171 @@
+```
+  k0s:
+    version: 0.11.0
+```
+
+```
+$ k0sctl apply --config k0sctl.yaml 
+
+⠀⣿⣿⡇⠀⠀⢀⣴⣾⣿⠟⠁⢸⣿⣿⣿⣿⣿⣿⣿⡿⠛⠁⠀⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠀█████████ █████████ ███
+⠀⣿⣿⡇⣠⣶⣿⡿⠋⠀⠀⠀⢸⣿⡇⠀⠀⠀⣠⠀⠀⢀⣠⡆⢸⣿⣿⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀███          ███    ███
+⠀⣿⣿⣿⣿⣟⠋⠀⠀⠀⠀⠀⢸⣿⡇⠀⢰⣾⣿⠀⠀⣿⣿⡇⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠀███          ███    ███
+⠀⣿⣿⡏⠻⣿⣷⣤⡀⠀⠀⠀⠸⠛⠁⠀⠸⠋⠁⠀⠀⣿⣿⡇⠈⠉⠉⠉⠉⠉⠉⠉⠉⢹⣿⣿⠀███          ███    ███
+⠀⣿⣿⡇⠀⠀⠙⢿⣿⣦⣀⠀⠀⠀⣠⣶⣶⣶⣶⣶⣶⣿⣿⡇⢰⣶⣶⣶⣶⣶⣶⣶⣶⣾⣿⣿⠀█████████    ███    ██████████
+
+INFO[0000] k0sctl 0.5.0 Copyright 2021, Mirantis Inc.   
+INFO[0000] Anonymized telemetry will be sent to Mirantis. 
+INFO[0000] By continuing to use k0sctl you agree to these terms: 
+INFO[0000] https://k0sproject.io/licenses/eula          
+INFO[0000] ==> Running phase: Connect to hosts                   
+INFO[0000] [ssh] 10.0.0.17:22: connected              
+INFO[0000] [ssh] 10.0.0.23:22: connected            
+INFO[0000] ==> Running phase: Detect host operating systems 
+INFO[0000] [ssh] 10.0.0.23:22: is running CentOS Linux 7 (Core) 
+INFO[0000] [ssh] 10.0.0.17:22: is running CentOS Linux 7 (Core) 
+INFO[0000] ==> Running phase: Prepare hosts   
+INFO[0000] ==> Running phase: Gather host facts 
+INFO[0000] [ssh] 10.0.0.23:22: discovered bond0 as private interface      valid_lft forever preferred_lft forever 
+INFO[0000] [ssh] 10.0.0.23:22: discovered 10.12.18.133 as private address 
+INFO[0000] ==> Running phase: Validate hosts   
+INFO[0000] ==> Running phase: Gather k0s facts 
+INFO[0000] [ssh] 10.0.0.23:22: found existing configuration                 
+INFO[0000] [ssh] 10.0.0.23:22: is running k0s controller version 0.10.0 
+INFO[0000] [ssh] 10.0.0.23:22: need upgrade: true                     
+INFO[0000] [ssh] 10.0.0.17:22: is running k0s worker version 0.10.0 
+INFO[0000] [ssh] 10.0.0.17:22: need upgrade: true   
+INFO[0000] [ssh] 10.0.0.23:22: checking if worker wrk has joined 
+INFO[0001] ==> Running phase: Configure K0s    
+INFO[0001] ==> Running phase: Upgrade controllers 
+INFO[0001] [ssh] 10.0.0.23:22: starting upgrade
+INFO[0001] [ssh] 10.0.0.23:22: Running with legacy service name, migrating... 
+INFO[0011] [ssh] 10.0.0.23:22: waiting for the k0s service to start 
+INFO[0016] ==> Running phase: Upgrade workers  
+INFO[0016] Upgrading 1 workers in parallel              
+INFO[0016] [ssh] 10.0.0.17:22: upgrade starting  
+INFO[0027] [ssh] 10.0.0.17:22: waiting for node to become ready again 
+INFO[0027] [ssh] 10.0.0.17:22: upgrade successful   
+INFO[0027] ==> Running phase: Disconnect from hosts 
+INFO[0027] ==> Finished in 27s                 
+INFO[0027] k0s cluster version 0.11.0-beta.2 is now installed 
+INFO[0027] Tip: To access the cluster you can now fetch the admin kubeconfig using: 
+INFO[0027]      k0sctl kubeconfig 
+```
+
+
+### Gotchas
+
+* upgrading k0s cluster with k0sctl is only possible if the cluster was already deployed using k0sctl
+* k0sctl is not able to detect exiting setup e.g. number of cluster nodes and similar
+* k0sctl can only add more nodes to the cluster but cannot remove them
```

Comment:
```suggestion
* k0sctl can currently only add more nodes to the cluster but cannot remove existing ones
```

With the drain/uncordon stuff implemented for upgrade this should be rather easy though.